### ISSUE TITLE
Support for Web Power Switch is missing

### ISSUE THREAD

**Issue by wwahammy** [qca/boardfarm#11]
Currently there's no support for Web Power Switch from DLI as a Power Device (   http://www.amazon.com/Power-Switch-DLI-Built--Easy-/dp/B00EZWD146/ref=sr_1_1?ie=UTF8&qid=1450381267&sr=8-1&keywords=web+power+switch). The title of the Web Power Switch login page is checked for [in Power.py](https://github.com/qca/boardfarm/blob/master/devices/power.py#L21) but the result isn't used.

**Comment by mbanders** [comment]
Use this library: https://pypi.python.org/pypi/dlipower/

It's works like this:

``` python
import dlipower
switch = dlipower.PowerSwitch(hostname="10.0.0.7", userid="admin")
# Turn off second outlet
switch.off(2)
# Turn off second outlet
switch.on(2)
```

You could add a WebPowerSwitch class in devices/power.py.

It would probably look something like this:

``` python
class WebPowerSwitch(PowerDevice):
    def __init__(self, ip_addr, username, password=None):
        self.switch = dlipower.PowerSwitch(hostname=ip_addr, userid=username)
    def reset(self, outlet):
        self.switch.off(outlet)
        time.sleep(5) # just so board capacitors empty or something
        self.switch.on(outlet)
```

If I had time I'd add it myself.

**Comment by wwahammy** [comment]
Sorry, I should have updated this. I have the support in my fork. I'll send a pull request for it in the next few days.

**Comment by mbanders** [comment]
No problem.  Another person was asking me about this very thing, and luckily you had posted an issue discussing it!

**Comment by iniesC** [comment]
@eric, can I expect this one in next few days? while I am using michael's change in my local repo.

**Comment by wwahammy** [comment]
@iniesC absolutely, I should have it submitted today. It's almost exactly Mike's fix though :)

**Comment by wwahammy** [comment]
@iniesC I submitted the changes in #35 . Check it out and let me know if it works for you.

**Comment by iniesC** [comment]
ok great. I see the code and looks right. I will test it out tomorrow. Have you merged it to master branch?

**Comment by wwahammy** [comment]
@iniesC: nope, haven't merged yet. I figured it'd be good to have someone else test to verify that it's working.

**Comment by iniesC** [comment]
@ericschultz This code doesn't work due to change in the signature of the 'get_power_device' function. For example, the call from openwrt_router (line 64) assumes that only ip address and outlet will be supplied. This results in outlet getting a default value None and dli module throws exception on reset. I like the signature change but that would require wide range of changes.

**Comment by wwahammy** [comment]
@iniesC sorry about that, I missed a few changes in bft and openwrt_router. Please let me know if the changes I've committed fixes that. I apologize for not being able to test them ahead of time; I'm so deep in getting my system to work other ways that just running at this commit wouldn't work.

**Comment by iniesC** [comment]
@ericschultz - No problem.

Could we keep the signature of the below classes consistent?
DLIPowerSwitch, SentrySwitchedCDU, APCPower

**Comment by wwahammy** [comment]
@iniesC sure, we could. In what way were you thinking?

**Comment by iniesC** [comment]
@ericschultz, If you are instantiating dlipowersitch class with username and password then could we also have similar signature for the other two classes? Right now for example, APCPower defaults to username and password within the class but for dli we pass in the value to the class.

**Comment by wwahammy** [comment]
@iniesC it makes sense to me. I think they were done that way because there was a thought that we wanted a default somewhere but it doesn't have to be in the function def. I don't know how standard the given defaults are outside of QCA though so it might not make sense to keep them at all.

**Comment by wwahammy** [comment]
@iniesC we could merge the current pull request and then I'd support you submitting a pull request making the functions consistent if you'd like

**Comment by wwahammy** [comment]
Closed by #35
